R/annotation.r

```diff
@@ -46,15 +46,22 @@ annotate <- function(geom, x = NULL, y = NULL, xmin = NULL, xmax = NULL,
 
   # Check that all aesthetic have compatible lengths
   lengths <- vapply(aesthetics, length, integer(1))
-  unequal <- length(unique(setdiff(lengths, 1L))) > 1L
-  if (unequal) {
+
+  # To determine the final number of rows `n` in the data frame,
+  # we need to find the unique lengths not equal to 1L (and there
+  # should be at most one such length). However, if all lengths
+  # are equal to 1L, then the final number of rows is also 1L.
```

Comment:
Maybe this would be a clearer approach?

```R
ulength <- unique(lengths)
unequal <- length(ulength) > 1 || ulength != 1
```

Reply:
No, that doesn't capture it. `unequal` should only be true if there is more than one length *excluding* 1. So if `lengths <- c(1, 2)`, we should have `unequal <- FALSE`, but if `lengths <- c(2, 3)`, we should have `unequal <- TRUE`. I'll see if I can find a better way to code this.

The original statement to calculate `unequal` was exactly right:
> `unequal <- length(unique(setdiff(lengths, 1L))) > 1L`
But now I'm also trying to capture the length `n` in addition, that required the additional code.

Reply:
I've thought about it some more, and I can't come up with something that would be clearer and simpler than the code in the current PR. Again, we also need to capture `n`, since it's used in line 64:
https://github.com/tidyverse/ggplot2/blob/4ff8243cf0215e5598045811dfc2ac94cd9a39ee/R/annotation.r#L64

Reply:
When I woke up this morning, I realized that using `setdiff()` only when needed may make the code easier to understand. I pushed a new version.